astroid/scoped_nodes.py

```diff
@@ -2701,6 +2716,16 @@ def get_children(self):
         if self.decorators is not None:
             yield self.decorators
 
+    def _get_assign_nodes(self):
+        for child_node in self.body:
+            yield from child_node._get_assign_nodes()
+
+    def _get_yield_nodes_skip_lambdas(self):
```

Comment:
Do you think it would make sense to make some of these duplicated functions into Mixins? I see `_get_yield_nodes_skip_lamdas` is placed exactly the same here twice